Closing Keyboard when clicked in the background

cdhiraj40 opened the pull request:
<!-- Please make sure that you have read our contribution guidelines: https://github.com/AntennaPod/AntennaPod/blob/develop/CONTRIBUTING.md#submit-a-pull-request -->

cdhiraj40 commented:
@ByteHamster as you can see that keyboard closing works perfectly in all the fragments which have parent Activity ```Main Activity.java```. 
now the keyboard opening part I think I could use some help from you as after my approach it is opening but also automatically closing just after getting opened, I don't know whats the issue

ByteHamster commented:
> now the keyboard opening part I think I could use some help from you as after my approach it is opening but also automatically closing just after getting opened, I don't know whats the issue

You are opening it before the search fragment is even shown. Shouldn't this be done in SearchFragment instead?

cdhiraj40 commented:
> > now the keyboard opening part I think I could use some help from you as after my approach it is opening but also automatically closing just after getting opened, I don't know whats the issue
> 
> You are opening it before the search fragment is even shown. Shouldn't this be done in SearchFragment instead?

Thanks for the help, now it works perfectly!
I have added support for onlineSearch and search fragment, you can check the recording (I have updated).
Happy hacking :)

cdhiraj40 commented:
> Thanks for working on this. I just tried this on my device and it seems to work fine. However, I think it is a bit of an overkill to check for hiding the keyboard every single time the app is touched, no matter where. It should be enough to do this in a scroll listener of the search result list.

Do you mean after the results have been shown? I will try to work on it, any approaches from you which you would like to say (like any functions already there which tell that results have been shown? would be appreciated :)

cdhiraj40 commented:
(Also, please fix the check style violations shown by the CI afterward)
 sure

cdhiraj40 commented:
> > any approaches from you which you would like to say (like any functions already there which tell that results have been shown?
> 
> I think it should be enough to register a scroll or touch listener to the recyclerView showing the list results. Not 100% sure how to do it but I'm pretty sure it is possible.
> 
so I tried with simple approach ``` setOnTouchListener ``` working fine :)

keunes commented:
@cdhiraj40 @ByteHamster Maybe this is out of scope, but I noticed that the keyboard doesn't open when opening the search fragment (tap on magnifier in e.g. the Queue screen), even though the search field is activated. I'm on 2.4.0-beta2. Could that be part of this improvement?

cdhiraj40 commented:
> @cdhiraj40 @ByteHamster Maybe this is out of scope, but I noticed that the keyboard doesn't open when opening the search fragment (tap on magnifier in e.g. the Queue screen), even though the search field is activated. I'm on 2.4.0-beta2. Could that be part of this improvement?

do you mean before this PR or after?
if after then I have added code for it which has covered it :)

cdhiraj40 commented:
> > do you mean before this PR or after?
> > if after then I have added code for it which has covered it :)
> 
> The issue exists before this PR (as I'm on the beta channel and this PR isn't even merged yet). Super, thanks for fixing it! <3

That's what I like to do :)

ByteHamster commented:
Just pushed a change that uses scroll listeners instead of touch listeners. Otherwise, the keyboard hiding code is called multiple times each second while scrolling

ByteHamster commented:
Thanks!
